cpp/include/graph.hpp

```diff
@@ -661,4 +663,179 @@ template <typename ET>
 struct invalid_edge_id : invalid_idx<ET> {
 };
 
+namespace mg {
+/**
+ * @brief              A Replicatable Graph representation to replicate
+ *                     a graph using the handle
+ *
+ * @tparam VT              Type of vertex id
+ * @tparam ET              Type of edge id
+ * @tparam WT              Type of weight
+ * @tparam GraphTypeView   Type of Graph View
+ */
+template <typename VT, typename ET, typename WT, typename GraphTypeView>
+class ReplicatableGraph {
+ public:
+  /**
+   * @brief                           Allows broadcast of a Graph via communicator and handle
+   * required memory
+   *
+   * @param[in] handle                Library handle (RAFT). If a communicator is expected
+   * to be initialized already
+   * @param[in]  graph_to_replicate  Pointer to the graph, should be null if rank != 0
+   * != 0
+   */
+  ReplicatableGraph(const raft::handle_t &handle, GraphTypeView const *graph_to_replicate)
+  {
+    handle_   = &handle;
+    rank_     = handle_->get_comms().get_rank();
+    has_data_ = false;
+    if (graph_to_replicate != nullptr) { graph = *graph_to_replicate; }
+  }
+
+  virtual void replicate()
+  {
+    replicate_info();
+    if (rank_ != 0) { initialize_data(); }
+    replicate_data();
+  }
+
+  virtual void replicate_info()
+  {
+    initialize_info_storage();
+    if (rank_ == 0) { fill_info_storage(); }
+    handle_->get_comms().bcast(d_buffer_.data().get(), d_buffer_.size(), 0, handle_->get_stream());
+    if (rank_ != 0) { read_info_storage(); }
+  }
+
+  virtual void initialize_info_storage()
+  {
+    size_t info_size = get_required_size_for_info();
```

Comment:
What's the purpose of the code above instead of
```
size_t info_size = sizeof(GraphTypeView);
```
?

Are you considering that the required size can actually deviate from sizeof(GraphTypeView) in the future?

I generally advise against creating 2-3 lines functions that are used only once. This goes against readability as one needs to jump between multiple locations (if that short routine is widely re-used, it is a different story, e.g. something like std::max, std::max(a,b) is more readable than a > b ? a : b).

Reply:
The function used to be longer but the final version simply used `sizeof(GraphTypeView)`, the function has been removed in eab569b